doc/nrf/app_dev/programming.rst

```diff
@@ -21,6 +21,41 @@ To program the :ref:`output build files <app_build_output_files>` to your device
 The flash command programs all cores by default, both in the |nRFVSC| and on the command line.
 If you want to program only one selected core, use ``west flash`` on the command line and :ref:`specify the domain <zephyr:west-multi-domain-flashing>`.
 
+.. _programming_selecting_runner:
+
+Selecting west runner
+*********************
+
+Starting with the |NCS| v3.0.0, all Nordic Semiconductor :ref:`boards <app_boards>` are using the `nRF Util`_ as the default runner for the ``west flash`` and ``west debug`` command.
+This change ensures that the flashing process is consistent across all boards.
+See the :ref:`build system section in the v3.0.0 migration guide <migration_3.0_recommended>` for more information.
+
+Runners are Zephyr-specific Python classes that wrap Zephyr's :ref:`flash and debug host tools <zephyr:flash-debug-host-tools>` and integrate them with west and the Zephyr build system to support ``west flash`` and related commands.
+You can read more about them in the Zephyr documentation in the :ref:`zephyr:flash-and-debug-support` and :ref:`zephyr:west-flashing` sections.
+
+If you want to change the runner used by ``west flash``, you can use one of the following options:
+
+* Specify the runner permanently in your application's :file:`CMakeLists.txt` file:
+
+  .. code-block::
+
+     set(BOARD_FLASH_RUNNER nrfjprog)
+
+* Specify the runner for a single command:
+
+  .. code-block::
+
+     west flash --r nrfjprog
+
+In either case, make sure you have the required tools installed on your system: `nRF Util`_ for ``-r nrfutil`` or the archived `nRF Command Line Tools`_ for ``-r nrfjprog``.
+
+To see which runners are available for your board, run the following command:
```

Comment:
the `--context` command outputs a lot of text.

Perhaps you want to highlight the bit that is interesting:

```
available runners in runners.yaml:
  nrfutil, nrfjprog, jlink, pyocd, openocd
default runner in runners.yaml:
  nrfutil
```